internal/pkg/workspace/workspace.go

```diff
@@ -185,26 +185,21 @@ func (ws *Workspace) workloadNames(match func(string) bool) ([]string, error) {
 	return names, nil
 }
 
-// ReadServiceManifest returns the contents of the service's manifest under copilot/{name}/manifest.yml.
-func (ws *Workspace) ReadServiceManifest(name string) ([]byte, error) {
-	mf, err := ws.readWorkloadManifest(name)
+// ReadWorkloadManifest returns the contents of the workload's manifest under copilot/{name}/manifest.yml.
+func (ws *Workspace) ReadWorkloadManifest(mftDirName string) (WorkloadManifest, error) {
+	raw, err := ws.read(mftDirName, manifestFileName)
 	if err != nil {
-		return nil, fmt.Errorf("read service %s manifest file: %w", name, err)
+		return nil, err
 	}
-	return mf, nil
-}
-
-// ReadJobManifest returns the contents of the job's manifest under copilot/{name}/manifest.yml.
-func (ws *Workspace) ReadJobManifest(name string) ([]byte, error) {
-	mf, err := ws.readWorkloadManifest(name)
+	mft := WorkloadManifest(raw)
+	mftName, err := mft.workloadName()
 	if err != nil {
-		return nil, fmt.Errorf("read job %s manifest file: %w", name, err)
+		return nil, err
 	}
-	return mf, nil
-}
-
-func (ws *Workspace) readWorkloadManifest(name string) ([]byte, error) {
-	return ws.read(name, manifestFileName)
+	if mftName != mftDirName {
+		return nil, fmt.Errorf("name of the manifest and directory do not match")
```

Comment:
should we add the values here:
```suggestion
		return nil, fmt.Errorf("name of the manifest '%s' and directory '%s' do not match", mftName, mftDirName)
```